google/cloud/spanner/integration_tests/client_integration_test.cc

```diff
@@ -179,58 +179,62 @@ TEST_F(ClientIntegrationTest, TransactionRollback) {
 
   // Cannot use RunTransaction in this test because we want to call Rollback
   // explicitly.
-  auto txn = MakeReadWriteTransaction();
-  auto insert1 = client_->ExecuteSql(
-      txn, SqlStatement("INSERT INTO Singers (SingerId, FirstName, LastName) "
-                        "VALUES (@id, @fname, @lname)",
-                        {{"id", Value(3)},
-                         {"fname", Value("test-fname-3")},
-                         {"lname", Value("test-lname-3")}}));
-  EXPECT_STATUS_OK(insert1);
-  auto insert2 = client_->ExecuteSql(
-      txn, SqlStatement("INSERT INTO Singers (SingerId, FirstName, LastName) "
-                        "VALUES (@id, @fname, @lname)",
-                        {{"id", Value(4)},
-                         {"fname", Value("test-fname-4")},
-                         {"lname", Value("test-lname-4")}}));
-  EXPECT_STATUS_OK(insert2);
-
-  auto reader = client_->Read(txn, "Singers", KeySet::All(),
-                              {"SingerId", "FirstName", "LastName"});
-  EXPECT_STATUS_OK(reader);
-
   using RowType = Row<std::int64_t, std::string, std::string>;
   std::vector<RowType> returned_rows;
-  if (reader) {
+
+  Transaction txn = MakeReadOnlyTransaction();
+  for (auto start = std::chrono::steady_clock::now(),
```

Comment:
My version has a breadcrumb comment here of ...
```
  // We do not use RunTransaction() here as we want to call Rollback()
  // directly. This means we need to retry ABORTED calls ourselves.
```
and (along the lines of previous comments on the matter) my loop only `continue`s on `ABORTED` calls, ala ...
```
  if (!insert1 && insert1.status().code() == StatusCode::kAborted) continue;
  EXPECT_STATUS_OK(insert1);
```